Documentation/proposals/OneDotNetMSBuildProperties.md

```diff
@@ -0,0 +1,178 @@
+
+In most cases, developers would only use `$(GenerateApplicationManifest)`
+if they want to try the new features in "legacy" Xamarin.
+
+## Android Template
+
+The default Android project template would include:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net6.0-android</TargetFramework>
+    <ApplicationTitle>MyApp</ApplicationTitle>
+    <ApplicationId>com.companyname.myapp</ApplicationId>
+    <ApplicationVersion>1.0</ApplicationVersion>
+  </PropertyGroup>
+</Project>
+```
+
+`$(ApplicationVersionCode)` is computed by a math expression to
+translate versions up to 128.128.128.128 to a 32-bit integer.
+
+Removed from `AndroidManifest.xml` in the project template:
+
+* `/manifest/@android:versionCode="1"`
+* `/manifest/@android:versionName="1.0"`
+* `/manifest/@package="com.companyname.myapp"`
+* `/application/@android:label="MyApp"`
+
+All values could be added later to the `AndroidManifest.xml` and
+override the MSBuild properties.
+
+## iOS Template
+
+The default iOS project template would include:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net6.0-ios</TargetFramework>
+    <ApplicationTitle>MyApp</ApplicationTitle>
+    <ApplicationId>com.companyname.myapp</ApplicationId>
+    <ApplicationVersion>1.0</ApplicationVersion>
+  </PropertyGroup>
+</Project>
+```
+
+`$(ApplicationDisplayVersion)` can default to `$(ApplicationVersion)`
+when blank.
+
+Removed from `Info.plist` in the project template:
+
+* `CFBundleDisplayName`
+* `CFBundleIdentifier`
+* `CFBundleVersion`
+* `CFBundleShortVersionString`
+
+All values could be added later to the `Info.plist` and override the
+MSBuild properties.
+
+## Example
+
+You could setup a cross-platform solution in .NET 6 with:
+
+* `Hello/Hello.csproj` - `net6.0` shared code
+* `HelloAndroid/HelloAndroid.csproj` - `net6.0-android`
+* `HelloiOS/HelloiOS.csproj` - `net6.0-android`
+* `Hello.sln`
+* `Directory.Build.props`
+
+Where `Directory.Build.props` can be setup for both platforms at once
+with:
+
+```xml
+<Project>
+  <PropertyGroup>
+    <ApplicationTitle>Hello!</ApplicationTitle>
+    <ApplicationId>com.companyname.hello</ApplicationId>
+    <!-- Increment to 1.0.1, 1.0.2, etc. on each release -->
+    <ApplicationVersion>1.0.0</ApplicationVersion>
+  </PropertyGroup>
+</Project>
+```
+
+In this project, a developer would only need to increment
+`$(ApplicationVersion)` for each public release.
+
+## Future Properties
+
+In future iterations, we can consider additional MSBuild properties
+beyond `$(ApplicationTitle)`, `$(ApplicationId)`,
+`$(ApplicationVersion)`, `$(ApplicationVersionCode)`, and
+`$(ApplicationDisplayVersion)`.
+
+This is a list of additional properties that cover most of the
+property pages in Visual Studio:
+
+* `$(iOSMinimumOSVersion)` _iOS only_
+* `$(iOSLaunchScreen)` _iOS only_
+* `$(iOSDeviceFamily)` _iOS only_ - `iPhone`, `iPad`, or `Universal`
```

Comment:
We'll need `tvOS*`, `MacCatalyst*` and `macOS*` variants for some of these as well.

Reply:
@Redth do we need to worry about Windows (UWP/WPF/etc.)? I will need to update this to consider tvOS, Catalyst, and macOS.

Reply:
I renamed some of properties to be `$(Apple*)` instead of `$(iOS*)`, Most of this is the "future" section, so we don't have to come up with the final names yet.